apps/src/blockly/customBlocks/googleBlockly/mutators/behaviorDefMutator.js

```diff
@@ -0,0 +1,206 @@
+   * Backwards compatible serialization implementation.
+   * @returns XML storage element.
+   * @this {Blockly.Block}
+   */
+  mutationToDom: function () {
+    const container = Blockly.utils.xml.createElement('mutation');
+    const params = this.getProcedureModel().getParameters();
+    for (let i = 0; i < params.length; i++) {
+      const parameter = Blockly.utils.xml.createElement('arg');
+      const varModel = params[i].getVariableModel();
+      parameter.setAttribute('name', varModel.name);
+      parameter.setAttribute('varid', varModel.getId());
+      container.appendChild(parameter);
+    }
+
+    // Save whether the statement input is visible.
+    if (!this.hasStatements_) {
+      container.setAttribute('statements', 'false');
+    }
+    return container;
+  },
+
+  /**
+   * Parse XML to set static behavior id, used for shared behaviors.
+   * @param xmlElement XML storage element.
+   * @this {Blockly.Block}
+   */
+  domToMutation: function (xmlElement) {
+    this.behaviorId = xmlElement.nextElementSibling.getAttribute('id');
+  },
+
+  /**
+   * Returns the state of this block as a JSON serializable object.
+   * @returns The state of this block, eg the parameters and statements.
+   */
+  saveExtraState: function () {
+    const state = Object.create(null);
+    state['procedureId'] = this.getProcedureModel().getId();
+    state['behaviorId'] = this.behaviorId;
+
+    const params = this.getProcedureModel().getParameters();
+    if (!params.length && this.hasStatements_) return state;
+
+    if (params.length) {
+      state['params'] = params.map(p => {
+        return {
+          name: p.getName(),
+          id: p.getVariableModel().getId(),
+          // Ideally this would be id, and the other would be varId,
+          // but backwards compatibility :/
+          paramId: p.getId(),
+        };
+      });
+    }
+    if (!this.hasStatements_) {
+      state['hasStatements'] = false;
+    }
+    return state;
+  },
+
+  /**
+   * Applies the given state to this block.
+   * @param state The state to apply to this block, eg the parameters and
+   *     statements.
+   */
+  loadExtraState: function (state) {
+    this.behaviorId = state['behaviorId'];
+    const map = this.workspace.getProcedureMap();
+    const procedureId = state['procedureId'];
+    if (
+      procedureId &&
+      procedureId !== this.model_.getId() &&
+      map.has(procedureId) &&
+      (this.isInsertionMarker() || this.noBlockHasClaimedModel_(procedureId))
+    ) {
+      if (map.has(this.model_.getId())) {
+        map.delete(this.model_.getId());
+      }
+      this.model_ = map.get(procedureId);
+    }
+
+    if (state['params'] && !this.getProcedureModel().getParameters().length) {
+      for (let i = 0; i < state['params'].length; i++) {
+        const {name, id, paramId} = state['params'][i];
+        this.getProcedureModel().insertParameter(
+          new ObservableParameterModel(this.workspace, name, paramId, id), //, type),
```

Comment:
@molly-moen Is your read based on this link that I can remove the `//, type),` piece here because we'll never need it? https://github.com/google/blockly-samples/blob/19ca60e2b8b6b60ba4d191cf42c42fc11249536f/plugins/block-shareable-procedures/src/observable_parameter_model.ts#L28

Reply:
that seems right to me